lib/Demangling/NodePrinter.cpp

```diff
@@ -3159,6 +3160,75 @@ void NodePrinter::printEntityType(NodePointer Entity, NodePointer type,
   }
 }
 
+std::string Demangle::keypathSourceString(const char *MangledName,
```

Comment:
It might be a little more readable if inside this you used a switch to check if the node is subscript/variable:
```cpp
switch (child->getKind()) {
case Node::Kind::Subscript: {
  // stuff here
  break;
}

case Node::Kind::Variable: {
  // stuff here
  break;
}

default:
  break;
}
```